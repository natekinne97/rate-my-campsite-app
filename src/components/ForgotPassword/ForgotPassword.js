import React from 'react';
import AuthApiService from '../../services/auth-api-service';


class ForgotPassword extends React.Component{

    state={
        sent: false,
        wait: null,
        error: ''
    }

    handleSubmit =  e =>  {
        e.preventDefault();
        const {email} = e.target;
        email.value = ''
        this.setState({
            wait: 'May take a moment',
            error: null
        })
        AuthApiService.emailReset({email: email.value})
            .then(res=>{
                this.setState({
                    sent: true
                })
            }).catch(error=>{
                this.setState({
                    sent: false,
                    error: error
                })
            })

    }

    render(){
        const sentMessage = "Email sent. Please follow the link for to reset your password.";
        return(
            <div className="gen-form">
                {this.state.sent
                ? <p>{sentMessage}</p>
                : null}
                {/* display feedback for long wait */}
                {this.state.wait
                ? <p>{this.state.wait}</p>
                : null}

                <form onSubmit={this.handleSubmit}>
                    <label htmlFor="email">Email: </label>
                    <input name="email" type="text"/>
                    <button type="submit">Recover password</button>
                </form>

            </div>
        )
    }

}
export default ForgotPassword;