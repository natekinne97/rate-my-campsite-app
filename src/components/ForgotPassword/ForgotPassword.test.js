import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ForgotPassword from './ForgotPassword';

describe.only('Review form component', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<ForgotPassword />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<ForgotPassword />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders the UI as expected with no unreads', () => {
        const tree = renderer
            .create(<ForgotPassword />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});