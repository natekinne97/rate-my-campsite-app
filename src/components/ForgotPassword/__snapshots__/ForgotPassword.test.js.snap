// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review form component renders the UI as expected 1`] = `
<div
  className="gen-form"
>
  <form
    onSubmit={[Function]}
  >
    <label
      htmlFor="email"
    >
      Email: 
    </label>
    <input
      name="email"
      type="text"
    />
    <button
      type="submit"
    >
      Recover password
    </button>
  </form>
</div>
`;

exports[`Review form component renders the UI as expected with no unreads 1`] = `
<div
  className="gen-form"
>
  <form
    onSubmit={[Function]}
  >
    <label
      htmlFor="email"
    >
      Email: 
    </label>
    <input
      name="email"
      type="text"
    />
    <button
      type="submit"
    >
      Recover password
    </button>
  </form>
</div>
`;
